macast/ssdp.py

```diff
@@ -164,7 +164,10 @@ def ssdp_main_thread(self):
             cherrypy.engine.publish("app_notify", "Macast", "SSDP Can't start")
             threading.Thread(target=lambda: Setting.stop_service(), name="SSDP_STOP_THREAD").start()
             return
-        self.sock.settimeout(1)
+        if Setting.get(SettingProperty.Check_IP_Frequency, 5):
+            self.sock.settimeout(1)
+            # see @server/..if check_ip_freq:
+            # for same reason
```

Comment:
The socket of SSDP listens to `0.0.0.0:1900`. I don't understand what this has to do with the change of local IP.

Reply:
I originally made changes on the main branch,  some logic of the main branch is to restart the whole process of monitoring the multicast group anyway after a small period of time. 
Then I made the same(in my thought) changes to some keywords when switching to the dev-test branch, have not noticing that this module has been refactored. I ran it in my environment and found no abnormality, so I submitted it hastily. I am very sorry for this.

As for the effect of listening to INADDR_ANY, the local routing table actually allocates data packets, so the changes of ip and interface are transparent to the program. But routing changes also brings some uncertainty, at least on my device these uncertainties also cannot be fixed by rebinding even to a fixed new ip.

So the ip who's change needs to be monitored is a multicast group member becomes unavailable or a new one outside the group becomes available, The former can be handled by waiting for multicast messages with a timeout counter, but the later is diffcult to monitor passively as I can see. Active polling all interfaces as it is now is a way to go, but when there are no interfaces were expected to be added or deleted frequently or got offline for a long time, I think it is also a reasonable choice to add the very new interfaces only by restart manually.

Reply:
Thanks for your very detailed explanation.

The version on the dev_test branch will not restart SSDP repeatedly, but will restart when the IP is changed. If the changes of IP is not monitored, SSDP restart will not be triggered.

It seems safe to delete this part of the modification.